lib/rex/proto/http/web_socket.rb

```diff
@@ -130,9 +133,9 @@ def write(buf, length = nil)
         length = buf.length
         buf = on_data_write(buf)
         if @write_type == :binary
-          @websocket.put_wsbinary(buf)
+          @websocket.put_wsbinary(buf, {mask: @mask_write})
         elsif @write_type == :text
-          @websocket.put_wstext(buf)
+          @websocket.put_wstext(buf, {mask: @mask_write})
```

Comment:
I don't see where this value is actually being used anywhere. It's being passed down through the write methods but it's not a keyword argument used by any of them. It looks like you're trying to pass it to `Frame.from_binary` / `Frame.from_text` but it's not making it there.

Also if I revert the changes in this file and make the necessary update in amazon_sms.rb:152, I can still open a session that works (other than the `exit` command).

Reply:
IMO we actually do want this - not every descendant of Channel will want to mask their writes, but as the API exists in master currently, they must. I might be missing some portage from my framework into the upstreaming one here, will dig in - thanks.

Reply:
Removed